AEPCore/Tests/MobileCoreTests.swift

```diff
@@ -107,14 +107,13 @@ class MobileCoreTests: XCTestCase {
         let expectation = XCTestExpectation(description: "unregistration completed in timely fashion")
         expectation.assertForOverFulfill = true
         MockExtension.unregistrationClosure = { expectation.fulfill() }
-        MobileCore.registerExtensions([MockExtension.self])
-
-        // test
-        MobileCore.unregisterExtension(MockExtension.self)
+        MobileCore.registerExtensions([MockExtension.self]) {
```

Comment:
This could again cause a failure if the test completes before registerExtensions finishes execution.
Can you modify as below to avoid race condition
```
let semaphore = DispatchSemaphore(value: 0)
MobileCore.registerExtensions([MockExtension.self]) {
    semaphore.signal()
}
semaphore.wait()

// test
MobileCore.unregisterExtension(MockExtension.self)
```